packages/cinterop/src/darwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1365,6 +1458,46 @@ actual object RealmInterop {
         )
     }
 
+    actual fun realm_set_add_notification_callback(
+        set: RealmSetPointer,
+        callback: Callback<RealmChangesPointer>
+    ): RealmNotificationTokenPointer {
+        return CPointerWrapper(
+            realm_wrapper.realm_set_add_notification_callback(
+                set.cptr(),
+                // Use the callback as user data
+                StableRef.create(callback).asCPointer(),
+                staticCFunction<COpaquePointer?, Unit> { userdata ->
+                    userdata?.asStableRef<Callback<RealmChangesPointer>>()?.dispose()
+                        ?: error("Notification callback data should never be null")
+                },
+                null, // See https://github.com/realm/realm-kotlin/issues/661
+                // Change callback
+                staticCFunction { userdata, change ->
+                    try {
+                        userdata?.asStableRef<Callback<RealmChangesPointer>>()?.get()?.onChange(
+                            CPointerWrapper(
+                                change,
+                                managed = false
+                            )
+                        ) // FIXME use managed pointer https://github.com/realm/realm-kotlin/issues/147
```

Comment:
It would be a better approach than copy the data into memory, list fine grained notifications don't use realm_clone because I didn't know about it at that time..

Reply:
Thanks for the heads up, I completely missed this one.